R/utilities-pk-parameter.R

```diff
@@ -0,0 +1,113 @@
+  userDefinedPKParameter <- UserDefinedPKParameter$new(name = name, standardPKParameter = standardPKParameter)
+
+  dimension <- NULL
+  if (!is.null(displayUnit)) {
+    dimension <- getDimensionForUnit(displayUnit)
+    if (is.null(dimension)) {
+      # we have a display unit and a dimension that is not found, this will be a user defiend dimension
+      dimension <- createUserDefinedDimension(displayUnit)
+    }
+  }
+  else {
+    dimension <- getDimensionForStandardPKParameter(standardPKParameter)
+  }
+
+  rClr::clrSet(userDefinedPKParameter$ref, "Dimension", dimension)
+  pkParameterTask <- getNetTask("PKParameterTask")
+  rClr::clrCall(pkParameterTask, "AddUserDefinedPKParameter", userDefinedPKParameter$ref)
+  .updatePKParameterProperties(userDefinedPKParameter, displayName, displayUnit)
+}
+
+#' @title Removes all User-Defined PK-Parameters that may have been added to the system
+#'
+#' @export
+removeAllUserDefinedPKParameters <- function() {
+  pkParameterTask <- getNetTask("PKParameterTask")
+  rClr::clrCall(pkParameterTask, "RemoveAllUserDefinedPKParameters")
+}
+
+#' @title Updates some properties of a PK-Parameter (displayName and displayUnit)
+#'
+#' @param name Name of PK-Parameter to update
+#' @param displayName Optional display name
+#' @param displayUnit Optional display unit. Note that the unit should be defined in unit of the dimension
+#'
+#' @examples
+#'
+#' updatePKParameter("t_max", "MyTmax", "min")
+#' @export
+updatePKParameter <- function(name, displayName = NULL, displayUnit = NULL) {
+  pkParameterTask <- getNetTask("PKParameterTask")
+  pkParameter <- pkParameterByName(name)
+
+  .updatePKParameterProperties(pkParameter, displayName, displayUnit)
+}
+
+.updatePKParameterProperties <- function(pkParameter, displayName = NULL, displayUnit = NULL) {
+  if (!is.null(displayName)) {
+    pkParameter$displayName <- displayName
+  }
+
+  if (!is.null(displayUnit)) {
+    pkParameter$displayUnit <- displayUnit
+  }
+
+  return(pkParameter)
+}
+
+#' @title Returns an instance of a PK-Parameter by name or NULL if the parameter by name is not found
+#'
+#' @param name Name of PK-Parameter to update
+#' @param stopIfNotFound Indicates whether an exception is thrown when the PK-Parameter was not found. Default is \code{TRUE}
+#'
+#' @examples
+#'
+#' pkParameter <- pkParameterByName(name = "t_max")
+#' @export
+pkParameterByName <- function(name, stopIfNotFound = TRUE) {
+  pkParameterTask <- getNetTask("PKParameterTask")
+  pkParameter <- rClr::clrCall(pkParameterTask, "PKParameterByName", name)
+  pkParameter <- toObjectType(pkParameter, PKParameter)
+
+  if (!is.null(pkParameter) || !stopIfNotFound) {
+    return(pkParameter)
+  }
+
+  stop(messages$errorPKParameterNotFound(name, allPKParameterNames()))
+}
+
+
+#' @title Returns the name of all pk parameters defined in the system
+#'
+#' @examples
+#'
+#' pkParameterNames <- allPKParameterNames()
+#' @export
+allPKParameterNames <- function(){

```

Comment:
Another usefull function @abdullahhamadeh : It returns all PK PArameter Names defined in the system. 
This is what I meant with maybe we do not need an enum . You could verify whatever inpuyt the uer is giving by checking if it is in this array or not

Reply:
Looks good 👍 
So the pk parameters and SA will be calculated for all elements returned by this?

Reply:
Correct. This is how it should work. You don't have to worry anymore about PKparameter type (user defined or not)

At least... that's the theory ;-)

Reply:
I have this list
![image](https://user-images.githubusercontent.com/18409098/76709379-aecd2000-66d4-11ea-80ba-3fb88e7b08c8.png)

but the pk calculation was done for this subset of the above:
![image](https://user-images.githubusercontent.com/18409098/76709403-d58b5680-66d4-11ea-9847-00db2c521c00.png)